client/wildcard/src/components/Button/Button.tsx

```diff
@@ -48,21 +53,49 @@ export interface ButtonProps
  */
 export const Button = React.forwardRef(
     (
-        { children, as: Component = 'button', type = 'button', variant, size, outline, className, ...attributes },
+        {
+            children,
+            as: Component = 'button',
+            type = 'button',
+            variant,
+            size,
+            outline,
+            className,
+            disabled,
+            ...attributes
+        },
         reference
-    ) => (
-        <Component
-            ref={reference}
-            className={classNames(
-                'btn',
-                variant && getButtonStyle({ variant, outline }),
-                size && getButtonSize({ size }),
-                className
-            )}
-            type={Component === 'button' ? type : undefined}
-            {...attributes}
-        >
-            {children}
-        </Component>
-    )
+    ) => {
+        const tooltip = attributes['data-tooltip']
+
+        const buttonComponent = (
+            <Component
+                ref={reference}
+                className={classNames(
+                    'btn',
+                    variant && getButtonStyle({ variant, outline }),
+                    size && getButtonSize({ size }),
+                    className
+                )}
+                type={Component === 'button' ? type : undefined}
+                disabled={disabled}
+                {...attributes}
+            >
+                {children}
+            </Component>
+        )
+
+        // Disabled elements don't fire mouse events, but the `Tooltip` relies on mouse
+        // events. This restores the tooltip behavior for disabled buttons by rendering an
+        // invisible `div` with the tooltip on top of the button, in the case that it is
+        // disabled. See https://stackoverflow.com/a/3100395 for more.
+        return disabled && tooltip ? (
+            <div className={styles.container}>
+                <div className={styles.disabledTooltip} data-tooltip={tooltip} />
+                {buttonComponent}
+            </div>
+        ) : (
+            buttonComponent
+        )
```

Comment:
nit: what do you think about using early returns instead of ternary here?

```suggestion
        if (disabled && tooltip) {
          return (
            <div className={styles.container}>
                <div className={styles.disabledTooltip} data-tooltip={tooltip} />
                {buttonComponent}
            </div>
        )
        
        return buttonComponent
```

Reply:
I like it! Thanks!